src/brsTypes/components/RoSGNode.ts

```diff
@@ -119,6 +121,25 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         this.parent = BrsInvalid.Instance;
     }
 
+    findNodeById(node: RoSGNode, id: BrsString): RoSGNode | BrsInvalid {
+        // test current node in tree
+        let currentId = node.get(new BrsString("id"));
+        if (currentId.toString() === id.toString()) {
+            return node;
+        }
+
+        // visit each child
+        for (let child of node.children) {
```

Comment:
Isn't this DFS? I think it's called out in the docs that BFS is used.

Reply:
yes this is DFS. This implementation matches the behavior seen on device.